src/main/java/FileInspector.kt

```diff
@@ -0,0 +1,57 @@
+import java.io.BufferedReader
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Paths
+
+object FileInspector {
+    private const val FILEPATH_IS_DIRECTORY = 1
+    private const val FILE_EXISTS = 2
+    private const val FILE_NOT_EXIST = 3
+    private const val NO_ACCESS_TO_FILE = 4
+
+    fun writeStringIntoFile(filePath: String, str: String) {
+        val fileAvailabilityCode = isFileAvailable(filePath)
+        if (fileAvailabilityCode != FILEPATH_IS_DIRECTORY && fileAvailabilityCode != NO_ACCESS_TO_FILE) {
+            if (fileAvailabilityCode == FILE_NOT_EXIST) {
+                File(filePath).bufferedWriter().use { out ->
+                    out.write(str)
+                }
+            } else if (fileAvailabilityCode == FILE_EXISTS) {
```

Comment:
Suggestion: I'd merge the two ifs. By the way, thanks for keeping this in, I wouldn't have anything else to write otherwise.